src/build/genprod.scala

```diff
@@ -353,7 +352,7 @@ object ProductTwo extends Product(2)
 
 class Product(val i: Int) extends Group("Product") with Arity {
   val productElementComment = """
-  /** Returns the n-th projection of this product if 0 < n <= productArity,
+  /** Returns the n-th projection of this product if 0 <= n < productArity,
    *  otherwise throws an `IndexOutOfBoundsException`.
    *
    *  @param n number of the projection to be returned
```

Comment:
Hmm, wondering now if the component at zero-based index 7 is properly called the 7th or the 8th projection.

If it is called the 8th projection then the main comment needs to be,

   Returns the (n+1)-th projection of this product if 0 <= n < productArity,